### PR TITLE
feat: migrate swap/spot history

### PR DESCRIPTION
**Pull request by Rickk137**
https://sovryn.monday.com/boards/2218344956/pulses/2732074064

### CODE REVIEW

**Comment by Rickk137** [comment]
https://sovryn.monday.com/boards/2218344956/pulses/2732074064

File: src/app/containers/SwapHistory/index.tsx [hunk 1]
```diff
@@ -32,76 +24,46 @@ import { Pagination } from '../../components/Pagination';
...
+      transactionHash: item.transaction.id,
+    }));
+  }, [data, loading]);
 
-  const onPageChanged = data => {
-    const { currentPage, pageLimit } = data;
-    const offset = (currentPage - 1) * pageLimit;
-    setCurrentHistory(history.slice(offset, offset + pageLimit));
-  };
+  const onPageChanged = data => setPage(data.currentPage);
```
Comment (tiltom) [inline src/app/containers/SwapHistory/index.tsx:61]: Use `useCallback`.

File: src/app/containers/SwapHistory/index.tsx [hunk 2]
```diff
@@ -222,32 +182,40 @@ export function SwapHistory() {
...
-interface AssetProps {
-  data: AssetRowData;
+interface IAssetProps {
+  data: IAssetRowData;
   itemFrom: AssetDetails;
   itemTo: AssetDetails;
 }
 
-function AssetRow({ data, itemFrom, itemTo }: AssetProps) {
+const AssetRow: React.FC<IAssetProps> = ({ data, itemFrom, itemTo }) => {
```
Comment (tiltom) [inline src/app/containers/SwapHistory/index.tsx:209]: Extract it to a separate component (create `components` folder in `SwapHistory`).

File: src/app/containers/SwapHistory/index.tsx [hunk 3]
```diff
@@ -222,32 +182,40 @@ export function SwapHistory() {
...
   const { t } = useTranslation();
   const dollars = useCachedAssetPrice(itemTo.asset, Asset.USDT);
   const dollarValue = useMemo(() => {
```
Comment (tiltom) [inline src/app/containers/SwapHistory/index.tsx:212]: Use `useDollarValue` hook instead.

File: src/app/containers/SwapHistory/index.tsx [hunk 4]
```diff
@@ -266,7 +234,7 @@ function AssetRow({ data, itemFrom, itemTo }: AssetProps) {
         <AssetRenderer asset={itemFrom.asset} />
       </td>
       <td>
-        {numberFromWei(data.returnVal._fromAmount)}{' '}
+        {toNumberFormat(data.fromAmount, 4)}{' '}
         <AssetRenderer asset={itemFrom.asset} />
```
Comment (tiltom) [inline src/app/containers/SwapHistory/index.tsx:238]: Use `AssetValue` component.

File: src/app/containers/SwapHistory/index.tsx [hunk 5]
```diff
@@ -280,12 +248,12 @@ function AssetRow({ data, itemFrom, itemTo }: AssetProps) {
       </td>
       <td className="tw-hidden lg:tw-table-cell">
         <div>
-          {numberFromWei(data.returnVal._toAmount)}{' '}
+          {toNumberFormat(data.toAmount || 0, 8)}{' '}
           <AssetRenderer asset={itemTo.asset} />
```
Comment (tiltom) [inline src/app/containers/SwapHistory/index.tsx:252]: Use `AssetValue` component.

File: src/app/pages/SpotTradingPage/components/SpotHistory/AssetRow.tsx [hunk 6]
```diff
@@ -8,42 +8,40 @@ import { translations } from 'locales/i18n';
...
 
   const [dollarValue, dollarsLoading] = useGetProfitDollarValue(
-    itemTo.asset,
-    data.returnVal._toAmount,
+    assetTo.asset,
+    toAmount || '0',
   );
 
-  const order = getOrder(itemFrom.asset, itemTo.asset);
+  const order = getOrder(assetFrom.asset, assetTo.asset);
```
Comment (tiltom) [inline src/app/pages/SpotTradingPage/components/SpotHistory/AssetRow.tsx:44]: Use `useMemo`.

File: src/app/pages/SpotTradingPage/components/SpotHistory/AssetRow.tsx [hunk 7]
```diff
@@ -74,48 +70,53 @@ export const AssetRow: React.FC<IAssetRowProps> = ({
         </span>
       </td>
       <td>
-        {numberFromWei(data.returnVal._fromAmount)}{' '}
-        <AssetRenderer asset={itemFrom.asset} />
+        {toNumberFormat(fromAmount, 4)}{' '}
+        <AssetRenderer asset={assetFrom.asset} />
```
Comment (tiltom) [inline src/app/pages/SpotTradingPage/components/SpotHistory/AssetRow.tsx:74]: Use `AssetValue` component.

File: src/app/pages/SpotTradingPage/components/SpotHistory/index.tsx [hunk 8]
```diff
@@ -1,110 +1,66 @@
...
-  }, [account, getHistory, log, retry]);
-
-  const onPageChanged = useCallback(
-    data => {
-      const { currentPage, pageLimit } = data;
-      const offset = (currentPage - 1) * pageLimit;
-      setCurrentHistory(history.slice(offset, offset + pageLimit));
-    },
-    [history],
+      .filter(item => item);
```
Comment (tiltom) [inline src/app/pages/SpotTradingPage/components/SpotHistory/index.tsx:56]: What is this `filter` supposed to do? I don't think you need it.

File: src/app/pages/SpotTradingPage/components/SpotHistory/index.tsx [hunk 9]
```diff
@@ -1,110 +1,66 @@
...
+  }, [data, loading]);
+
+  const onPageChanged = data => setPage(data.currentPage);
```
Comment (tiltom) [inline src/app/pages/SpotTradingPage/components/SpotHistory/index.tsx:59]: Use `useMemo`.

File: src/app/pages/SpotTradingPage/components/SpotHistory/index.tsx [hunk 10]
```diff
@@ -123,26 +79,21 @@ export const SpotHistory: React.FC = () => {
...
-            key={item.transactionHash}
-            data={data}
-            itemFrom={assetFrom!}
-            itemTo={assetTo!}
-          />
-        );
-      });
+        return <AssetRow key={item.transactionHash} {...data} />;
+      })
+      .filter(item => item);
```
Comment (tiltom) [inline src/app/pages/SpotTradingPage/components/SpotHistory/index.tsx:96]: Again, I don't think you need it.

File: src/app/pages/SpotTradingPage/components/SpotHistory/index.tsx [hunk 11]
```diff
@@ -183,22 +134,16 @@ export const SpotHistory: React.FC = () => {
...
-                <AssetRow
-                  key={item.transaction_hash}
-                  data={item}
-                  itemFrom={assetFrom}
-                  itemTo={assetTo}
-                />
-              );
+            {items.map(item => {
+              if (!item) return null;
+              return <AssetRow key={item?.transactionHash} {...item} />;
```
Comment (tiltom) [inline src/app/pages/SpotTradingPage/components/SpotHistory/index.tsx:139]:
```suggestion
              if (!item) {
                  return null;
              }
              
              return <AssetRow key={item?.transactionHash} {...item} />;
```

Comment (tiltom) [reply src/app/pages/SpotTradingPage/components/SpotHistory/index.tsx:139]: No inline ifs.

**Review by tiltom — CHANGES_REQUESTED** [review]

File: src/app/pages/SpotTradingPage/components/SpotHistory/index.tsx (hunk 8 above)
Comment (Rickk137) [reply src/app/pages/SpotTradingPage/components/SpotHistory/index.tsx:56]: If assetFrom OR assetTo is not found from the token id then the item would be null and this filter would make sure to remove those items

File: src/app/pages/SpotTradingPage/components/SpotHistory/index.tsx (hunk 10 above)
Comment (Rickk137) [reply src/app/pages/SpotTradingPage/components/SpotHistory/index.tsx:96]: the same reason

File: src/app/containers/SwapHistory/index.tsx [hunk 12]
```diff
@@ -1,107 +1,58 @@
...
+  const [page, setPage] = useState(1);
+
+  const { data, loading } = useGetSwapHistoryQuery({
+    variables: { user: account.toLowerCase() },
+    pollInterval: APOLLO_POLL_INTERVAL,
+  });
+
+  const history = useMemo(() => {
+    if (loading || !data) return [];
+
```
Comment (tiltom) [inline src/app/containers/SwapHistory/index.tsx:39]:
```suggestion
    if (loading || !data) {
        return [];
    }

```

File: src/app/pages/SpotTradingPage/components/SpotHistory/index.tsx [hunk 13]
```diff
@@ -1,110 +1,66 @@
...
-            })
-            .filter(item => item),
+  const { data, loading } = useGetSwapHistoryQuery({
+    variables: { user: account.toLowerCase() },
+    pollInterval: APOLLO_POLL_INTERVAL,
+  });
+
+  const history = useMemo(() => {
+    if (loading || !data) return [];
+
```
Comment (tiltom) [inline src/app/pages/SpotTradingPage/components/SpotHistory/index.tsx:38]:
```suggestion
    if (loading || !data) {
        return [];
    }

```

File: src/app/pages/SpotTradingPage/components/SpotHistory/index.tsx [hunk 14]
```diff
@@ -123,26 +79,21 @@ export const SpotHistory: React.FC = () => {
         const assetTo = assets.find(
           currency => currency.asset === customData?.targetToken,
         );
+        if (!assetFrom || !assetTo) return null;
```
Comment (tiltom) [inline src/app/pages/SpotTradingPage/components/SpotHistory/index.tsx:83]:
```suggestion
        if (!assetFrom || !assetTo) {
             return null;
        }

```

**Review by tiltom — CHANGES_REQUESTED** [review]

**Review by tiltom — APPROVED** [review]

**Review by Alitasovryn — APPROVED** [review]

**Review by soulBit — APPROVED** [review]

**Pull request merged**